### ISSUE TITLE
[8.0] Unable to use a closed cursor

### ISSUE THREAD

**Issue by guewen** [odoo/odoo#6597]
**Unable to use a closed cursor**

Impacted versions:

 - 8.0, since commit 071152216fe44039f75e821b0cbf31344b66d174

Steps to reproduce:

I couldn't get the exact condition to reproduce it.
I get it when I run a `behave` scenario (with https://github.com/camptocamp/oerpscenario/).
It uses `erppeek` to interact with Odoo (with an `import openerp`).
The scenario does:
1. Create a database
2. Install some modules on the database
3. Load some files with the load() method

It achieves to import some files, and on a specific file it fails. I assume that is because the first files do not have to get the `digits` of a field.

Current behavior:

Traceback:

```
 Traceback (most recent call last):                                                                                                                                                                                 
   File "/home/gbaconnier/.buildout/eggs/behave-1.2.4-py2.7.egg/behave/model.py", line 1173, in run                                                                                                                 
     match.run(runner.context)                                                                                                                                                                                      
   File "/home/gbaconnier/.buildout/eggs/behave-1.2.4-py2.7.egg/behave/model.py", line 1589, in run                                                                                                                 
     self.func(context, *args, **kwargs)                                                                                                                                                                            
   File "/home/gbaconnier/dev/Scenario/OERPScenario/features/steps/tools.py", line 56, in impl                                                                                                                      
     result = model(model_name).load(head, values, ctx.oe_context)                                                                                                                                                  
   File "/home/gbaconnier/.buildout/eggs/ERPpeek-1.6.1-py2.7.egg/erppeek.py", line 1162, in wrapper                                                                                                                 
     return self._execute(attr, *params, **kwargs)                                                                                                                                                                  
   File "/home/gbaconnier/.buildout/eggs/ERPpeek-1.6.1-py2.7.egg/erppeek.py", line 693, in execute                                                                                                                  
     res = self._execute(obj, method, *params)                                                                                                                                                                      
   File "/home/gbaconnier/.buildout/eggs/ERPpeek-1.6.1-py2.7.egg/erppeek.py", line 383, in <lambda>                                                                                                                 
     wrapper = lambda s, *args: s._dispatch(name, args)                                                                                                                                                             
   File "/home/gbaconnier/dev/parts/server/openerp/http.py", line 114, in dispatch_rpc                                                                                                                              
     result = dispatch(method, params)                                                                                                                                                                              
   File "/home/gbaconnier/dev/parts/server/openerp/service/model.py", line 37, in dispatch                                                                                                                          
     res = fn(db, uid, *params)                                                                                                                                                                                     
   File "/home/gbaconnier/dev/parts/server/openerp/service/model.py", line 113, in wrapper                                                                                                                          
     return f(dbname, *args, **kwargs)                                                                                                                                                                              
   File "/home/gbaconnier/dev/parts/server/openerp/service/model.py", line 170, in execute                                                                                                                          
     res = execute_cr(cr, uid, obj, method, *args, **kw)                                                                                                                                                            
   File "/home/gbaconnier/dev/parts/server/openerp/service/model.py", line 159, in execute_cr                                                                                                                       
     return getattr(object, method)(cr, uid, *args, **kw)                                                                                                                                                           
   File "/home/gbaconnier/dev/parts/server/openerp/api.py", line 241, in wrapper                                                                                                                                    
     return old_api(self, *args, **kwargs)                                                                                                                                                                          
   File "/home/gbaconnier/dev/parts/server/openerp/models.py", line 1046, in load                                                                                                                                   
     fg = self.fields_get(cr, uid, context=context)                                                                                                                                                                 
   File "/home/gbaconnier/dev/parts/server/openerp/api.py", line 241, in wrapper                                                                                                                                    
     return old_api(self, *args, **kwargs)                                                                                                                                                                          
   File "/home/gbaconnier/dev/parts/server/openerp/models.py", line 3078, in fields_get                                                                                                                             
     description = field.get_description(recs.env)                                                                                                                                                                  
   File "/home/gbaconnier/dev/parts/server/openerp/fields.py", line 642, in get_description                                                                                                                         
     value = getattr(self, prop)                                                                                                                                                                                    
   File "/home/gbaconnier/dev/parts/server/openerp/fields.py", line 1060, in digits                                                                                                                                 
     return self._digits(cr)                                                                                                                                                                                        
   File "/home/gbaconnier/dev/parts/server/addons/account/account.py", line 1847, in change_digit_tax                                                                                                               
     res = openerp.registry(cr.dbname)['decimal.precision'].precision_get(cr, SUPERUSER_ID, 'Account')                                                                                                              
   File "/home/gbaconnier/dev/parts/server/openerp/api.py", line 241, in wrapper                                                                                                                                    
     return old_api(self, *args, **kwargs)                                                                                                                                                                          
   File "<string>", line 2, in precision_get                                                                                                                                                                        
   File "/home/gbaconnier/dev/parts/server/openerp/tools/cache.py", line 74, in lookup                                                                                                                              
     value = d[key] = self.method(*args, **kwargs)                                                                                                                                                                  
   File "/home/gbaconnier/dev/parts/server/addons/decimal_precision/decimal_precision.py", line 44, in precision_get                                                                                                
     cr.execute('select digits from decimal_precision where name=%s', (application,))                                                                                                                               
   File "/home/gbaconnier/dev/parts/server/openerp/sql_db.py", line 157, in wrapper                                                                                                                                 
     raise psycopg2.OperationalError(msg)                                                                                                                                                                           
 OperationalError: Unable to use a closed cursor. 
```

**Comment by guewen** [comment]
@rco-odoo do you have an idea what that could be? Could it be related to how `erppeek` initializes an environment? (https://github.com/tinyerp/erppeek/blob/master/erppeek.py#L268-L269)

**Comment by KangOl** [comment]
Should be fixed with 5efac22043adc735d9502fb9f6f7466b85916573 and 1b5576af4658b979b97976e23030a03c167d3bc8

**Comment by guewen** [comment]
Thanks!
